File: js/nova/time/Date.js
var Date = function () {
	var self = this;
	
	this.year = 0;
	this.month = 0;
	this.day = 0;
	this.hour = 0;
	this.minute = 0;
	this.second = 0;
	
	this.Date = function () {
		self.this();
	};
	
	this.destroy = function () {
	};
	
	this.this = function () {
		self.updateTime();
	};
	
	this.decodeDate = function (prototype, date) {
	};
	
	this.updateTime = function () {
		self.nova_updateTime();
		self.year = self.nova_getYear();
		self.month = self.nova_getMonth();
		self.day = self.nova_getDay();
		self.hour = self.nova_getHour();
		self.minute = self.nova_getMinute();
		self.second = self.nova_getSecond();
	};
	
	this.formatDate = function (str) {
		return self.formatDate(str, self.month, self.day, self.year, self.hour, self.minute, self.second);
	};
	
	this.formatDate = function (str, first, second, third, fourth, fifth, sixth) {
		var data;
		data = self.nova_formatDate(str.chars.data, first, second, third, fourth, fifth, sixth);
		return new String(data);
	};
	
	this.super = function () {
	};
	
	
};

File: java/nova/time/Date.java
package nova.time;

import java.util.Optional;
import nova.exception.ExceptionData;
import nova.exception.Exception;
import nova.exception.DivideByZeroException;
import nova.io.Console;
import nova.primitive.number.Number;
import nova.primitive.number.Byte;
import nova.primitive.number.Short;
import nova.primitive.number.Int;
import nova.primitive.number.Long;
import nova.primitive.number.Float;
import nova.primitive.number.Double;
import nova.primitive.Null;
import nova.primitive.number.Char;
import nova.primitive.Bool;
import nova.datastruct.list.Array;
import nova.datastruct.list.IntArray;
import nova.datastruct.list.CharArray;
import nova.datastruct.list.DoubleArray;
import nova.datastruct.list.IntRange;
import nova.thread.Thread;
import nova.thread.async.Async;
import nova.gc.GC;
import nova.math.Math;
import nova.Object;
import nova.String;
import nova.System;
import nova.Class;

public class NovaDate
{
	
	public int year;
	public int month;
	public int day;
	public int hour;
	public int minute;
	public int second;
	
	
	
	
	public NovaDate()
	{
		init();
	}
	
	
	
	
	
	
	
	
	
	public void init()
	{
		updateTime();
	}
	
	public void decodeDate(NovaString prototype, NovaString date)
	{
	}
	
	public void updateTime()
	{
		nova_updateTime();
		year = nova_getYear();
		month = nova_getMonth();
		day = nova_getDay();
		hour = nova_getHour();
		minute = nova_getMinute();
		second = nova_getSecond();
	}
	
	public NovaString formatDate(NovaString str)
	{
		return formatDate(str, month, day, year, hour, minute, second);
	}
	
	public NovaString formatDate(NovaString str, int first, int second, int third, int fourth, int fifth, int sixth)
	{
		char[] data;
		data = nova_formatDate(str.chars.data, first, second, third, fourth, fifth, sixth);
		return new NovaString(data);
	}
	
}
